test/endpoints/baz/baz_metrics_test.go

```diff
@@ -88,7 +88,7 @@ func TestCallMetrics(t *testing.T) {
 		return
 	}
 
-	cg.MetricsWaitGroup.Wait()
+	<-cg.MetricsWaitGroup.Wait
```

Comment:
channels should not be part of the public interface, they are an implementation detail.

Let's add a `Wait()` method that calls `<-cg.MetricsWaiter.waitCh`